src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -1775,27 +1808,47 @@ impl Default for SelectionMode {
 
 #[allow(unused_parens)]
 fn new_graph_editor(app:&Application) -> GraphEditor {
-    let world          = &app.display;
-    let scene          = world.scene();
-    let cursor         = &app.cursor;
-    let frp            = Frp::new();
-    let model          = GraphEditorModelWithNetwork::new(app,cursor.clone_ref(),&frp);
-    let network        = &frp.network;
-    let nodes          = &model.nodes;
-    let edges          = &model.edges;
-    let inputs         = &model.frp;
-    let mouse          = &scene.mouse.frp;
-    let touch          = &model.touch_state;
-    let visualizations = &model.visualizations;
-    let logger         = &model.logger;
-    let out            = &frp.output;
+    let world        = &app.display;
+    let scene        = world.scene();
+    let cursor       = &app.cursor;
+    let frp          = Frp::new();
+    let model        = GraphEditorModelWithNetwork::new(app,cursor.clone_ref(),&frp);
+    let network      = &frp.network;
+    let nodes        = &model.nodes;
+    let edges        = &model.edges;
+    let inputs       = &model.frp;
+    let mouse        = &scene.mouse.frp;
+    let touch        = &model.touch_state;
+    let vis_registry = &model.vis_registry;
+    let logger       = &model.logger;
+    let out          = &frp.output;
 
     // FIXME : StyleWatch is unsuitable here, as it was designed as an internal tool for shape system (#795)
     let styles             = StyleWatch::new(&scene.style_sheet);
     let any_type_sel_color = styles.get_color(theme::code::types::any::selection);
 
 
 
+    // ========================
+    // === Scene Navigation ===
+    // ========================
+
+    frp::extend! { network
+
+        no_vis_selected <- out.some_visualisation_selected.on_false();
+        some_vis_selected <- out.some_visualisation_selected.on_true();
+
+        disable_navigator <- any_(inputs.disable_navigator,some_vis_selected);
+        enable_navigator  <- any_(inputs.enable_navigator,no_vis_selected);
+
+        eval_ disable_navigator ( model.navigator.disable() );
+        eval_ enable_navigator  ( model.navigator.enable()  );
+
```

Comment:
If you'd use a single frp event instead of `inputs.disable_navigator` and `inputs.enable_navigator`, then you'd be able to write (new FRP syntax ahead!) :) 

```rust
navigator_active <- inputs.set_navigator_disabled || some_vis_selected;
```
please note that Im using `inputs.set_navigator_disabled` because FRP endpoints initilize with defaults, so this will initialize with false. If we would name it `inputs.set_navigator_active`, it would initialize with false, which would give bad value when queries when `some_vis_selected` will fire.